tests/v2/validation/provisioning/rke2/templates_test.go

```diff
@@ -0,0 +1,164 @@
+package rke2
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	v1 "github.com/rancher/rancher/pkg/apis/catalog.cattle.io/v1"
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/extensions/charts"
+	"github.com/rancher/shepherd/extensions/cloudcredentials"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/extensions/defaults"
+	"github.com/rancher/shepherd/extensions/projects"
+	"github.com/rancher/shepherd/extensions/provisioning"
+	"github.com/rancher/shepherd/extensions/users"
+	password "github.com/rancher/shepherd/extensions/users/passwordgenerator"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/rancher/shepherd/pkg/wait"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+const (
+	localCluster   = "local"
+	fleetNamespace = "fleet-default"
+)
+
+type ClusterTemplateTest struct {
```

Comment:
I would think this more so belongs in shepherd as opposed to the test file. Would be curious to know what @slickwarren or @Josh-Diamond thinks here.

Reply:
is the function shared among multiple packages? -> no, so it won't be an extension. But I think this should be moved into its own file and have a different name, maybe just `ClusterTemplate` in `cluster_template.go`

Reply:
I have added the clusterTemplateStruct to provisioninginput.go in shepherd.